src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultAssemblyPartDiscoveryProvider.cs

```diff
@@ -36,8 +36,12 @@ public static IEnumerable<ApplicationPart> DiscoverAssemblyParts(string entryPoi
         {
             var entryAssembly = Assembly.Load(new AssemblyName(entryPointAssemblyName));
             var context = DependencyContext.Load(entryAssembly);
-
-            return GetCandidateAssemblies(entryAssembly, context).Select(p => new AssemblyPart(p));
+            foreach (var assembly in GetCandidateAssemblies(entryAssembly, context))
+            {
+                var isEntryPoint = ReferenceEquals(entryAssembly, assembly);
+                var assemblyPart = new AssemblyPart(assembly, isEntryPoint);
+                yield return assemblyPart;
+            }
```

Comment:
Should we throw if there's more than one entry point?

Reply:
Yes